src/System.Runtime.Analyzers/Core/AvoidZeroLengthArrayAllocations.cs

```diff
@@ -83,6 +90,28 @@ private void AnalyzeOperation(OperationAnalysisContext context)
             }
         }
 
+        private static bool IsCompilerGeneratedParamsArray(IArrayCreationExpression arrayCreationExpression, OperationAnalysisContext context)
+        {
+            var model = context.Compilation.GetSemanticModel(arrayCreationExpression.Syntax.SyntaxTree);
+
+            // Compiler generated array creation seems to just use the syntax from the parent.
+            var parent = model.GetOperation(arrayCreationExpression.Syntax, context.CancellationToken) as IInvocationExpression;
```

Comment:
We should also consider exposing SemanticModel on OperationAnalysisContext. This may be null for scenarios where the Operation tree is not from source, but we can at least reuse the cached semantic model for source cases.